### PR TITLE
chore(Compiler Warnings): Use ssl:handshake instead of ssl:accept for…

### PR DESCRIPTION
**Pull request by tomciopp**
… OTP versions >= 21

@arjan I'm not 100% if this syntax is correct or if it can be simplified, I'm relying on syntax highlighting from my editor and other examples from the repo. I will be happy to fix if anything looks wrong.

### CODE REVIEW

File: src/smtp_socket.erl [hunk 1]
```diff
@@ -120,13 +120,24 @@ accept(Socket, Timeout) when is_port(Socket) ->
...
 	end.
+-else.
+accept(Socket, Timeout) ->
+	case ssl:transport_accept(Socket, Timeout) of
+		{ok, NewSocket} ->
+			ssl:handshake(NewSocket),
+			{ok, NewSocket};
+		Error -> Error
+	end.
+-endif.
```
Comment (seriyps) [inline src/smtp_socket.erl]: another option would be to create a new function and use it in both places:

```
-ifdef(deprecated_ssl_accept).
ssl_handshake(Sock) ->
    ssl:handshake(Sock).
-else.
ssl_handshake(Sock) ->
    ssl:ssl_accept(NewSocket).
-endif.
```
This way you will have less code duplication.

Comment (arjan) [reply src/smtp_socket.erl]: Yes I agree with @seriyps here, less code duplication = better

**Comment by mworrell** [comment]
@tomciopp We are preparing a 0.15 release, are you able to make the proposed changes and rebase?

**Comment by mworrell** [comment]
I think this will give a merge conflict with #171, so we can do two things:

 1. Add this to a separate 0.x branch (which kind of makes sense)
 2. Add this to the master branch and then rebase #171 

@seriyps I think option (1) makes most sense, what do you think?

**Comment by seriyps** [comment]
Well, conflict is not a problem for me. I can rebase / fix it, it's ok. So, up to you.

**Comment by tomciopp** [comment]
@mworrell I can take care of this later today, or early tomorrow morning.

**Comment by tomciopp** [comment]
@mworrell @seriyps @arjan Updated code so that there is a single function called for both situations. Let me know if anything else needs to change.

**Comment by mworrell** [comment]
Looks good to me as well. Merging.

Thanks @tomciopp!

File: src/smtp_socket.erl [hunk 2]
```diff
@@ -120,14 +120,30 @@ accept(Socket, Timeout) when is_port(Socket) ->
 		Error ->
 			Error
 	end;
+
+
 accept(Socket, Timeout) ->
 	case ssl:transport_accept(Socket, Timeout) of
 		{ok, NewSocket} ->
-			ssl:ssl_accept(NewSocket),
+			ssl_handshake(NewSocket),
```
Comment (lazedo) [inline src/smtp_socket.erl]: in OTP 21, handshake always returns a new Socket and we are dropping it here ?

Comment (mworrell) [reply src/smtp_socket.erl]: Thanks, see https://github.com/gen-smtp/gen_smtp/pull/187 if that would solve this.  (All tests pass, so apparently at the moment the ssl routines return the same socket)

**Pull request merged**